### PR TITLE
Removes exponetial numBonds behavior

### PR DESCRIPTION
**Pull request by bp-kelley**
As it turns out, boost::graph's use of a list has a non-constant behavior for getting the number of bonds (edges).  This is a patch to prevent this behavior for large systems.

It is somewhat informational, @coleb - please check to see if this helps your run-times.

### CODE REVIEW

File: Code/GraphMol/ROMol.h [hunk 1]
```diff
@@ -191,12 +194,18 @@ class ROMol : public RDProps {
...
     dp_ringInfo = 0;
     initFromOther(other, quickCopy, confId);
+    numBonds = rdcast<unsigned int>(boost::num_edges(d_graph));
   };
   //! construct a molecule from a pickle string
   ROMol(const std::string &binStr);
 
   virtual ~ROMol() { destroy(); };
 
+  //! \name Molecule Name or Title
```
Comment (bp-kelley) [inline Code/GraphMol/ROMol.h]: Oops - see what I snuck in here? :)

Comment (greglandrum) [inline Code/GraphMol/ROMol.h]: how on *earth* did that happen? ;-)

Comment (greglandrum) [inline Code/GraphMol/ROMol.h]: Jokes aside: I think if this is going to be here, then the name should logically be an attribute, not a property (for that we should confirm that it'd doesn't make the molecule too much bigger) and that it should definitely be pickled..

**Comment by coleb** [comment]
The performance improvement looks good on our end. Made a simple graph with the attached python to show the exponential improvement in AddBond performance only toggling the 7d4591c014d3966e9382a16938e3af5b365b0e8e commit. 

pullrequest1154performancegraph

[time_add_bond_on_long_chain.txt](https://github.com/rdkit/rdkit/files/586177/time_add_bond_on_long_chain.txt)

@greglandrum is right though, should rebase out the molecule title stuff.

**Comment by bp-kelley** [comment]
Yeah, the Get/SetName were pushed by accident, they have been removed.

**Comment by bp-kelley** [comment]
@greglandrum @coleb I think this is one of those cases where we can't really be elegant, Brian C and I tried to change from a list to a vector, but boost graph died hard.

When we fully go to C++14 we can probably remove this as list.size() will be constant time I think.

It's also one of those cases where having a good regression suite makes this kind of optimization easy...

**Pull request merged**